catalogue/webapp/utils/iiif/v3/canvas.ts

```diff
@@ -13,28 +17,45 @@ type Thumbnail = {
   }[];
 };
 
-export function getThumbnailImage(canvas: Canvas):
-  | {
-      width: number;
-      url: string | undefined;
-    }
-  | undefined {
+export function getThumbnailImage(canvas: Canvas): ThumbnailImage | undefined {
   if (!canvas.thumbnail) return;
-  const thumbnail = canvas.thumbnail[0] as Thumbnail; // ContentResource which this should be, doesn't have a service property
-  const thumbnailService = Array.isArray(thumbnail.service)
-    ? thumbnail.service[0]
-    : thumbnail.service;
-  const urlTemplate = iiifImageTemplate(thumbnailService['@id']);
-  const preferredMinThumbnailHeight = 400;
-  const preferredThumbnail = thumbnailService?.sizes
-    ?.sort((a, b) => a.height - b.height)
-    .find(dimensions => dimensions.height >= preferredMinThumbnailHeight);
-  return {
-    width: preferredThumbnail?.width || 30,
-    url:
-      urlTemplate &&
-      urlTemplate({
-        size: `${preferredThumbnail ? `${preferredThumbnail.width},` : 'max'}`,
-      }),
-  };
+
+  // The potential type here is extremely wide, but in practice we only
+  // see a couple of formats in our manifests:
+  //
+  //    - A thumbnail with an associated IIIF image service, which is used
+  //      for digitised books
+  //    - A thumbnail with type Image but no associated image service, which
+  //      is used for e.g. digitised PDFs
+  //
+  // We use our own type rather than trying to shoehorn this into the ContentResource
+  // type provided by the IIIF libraries.
+  //
+  // See the tests for examples of each of these.
+  const thumbnail = canvas.thumbnail[0] as Thumbnail;
+
+  if (isNotUndefined(thumbnail.service)) {
+    const thumbnailService = Array.isArray(thumbnail.service)
+      ? thumbnail.service[0]
+      : thumbnail.service;
+
+    const urlTemplate = iiifImageTemplate(thumbnailService['@id']);
+    const preferredMinThumbnailHeight = 400;
+    const preferredThumbnail = thumbnailService?.sizes
+      ?.sort((a, b) => a.height - b.height)
+      .find(dimensions => dimensions.height >= preferredMinThumbnailHeight);
+    return (
+      urlTemplate && {
```

Comment:
double checking my thought process, this could return `false`, but it should return `undefined` if there is no `urlTemplate`?

Reply:
I think it would return `undefined`, but actually I think this line is wrong – the conditional is unnecessary, we know that `urlTemplate` is defined at this point.